query/exprpb/proto.go

```diff
@@ -0,0 +1,591 @@
+		expr, err := ExprFromProto(e.Alias.Expr)
+		if err != nil {
+			return nil, err
+		}
+
+		return &logicalplan.AliasExpr{
+			Expr:  expr,
+			Alias: e.Alias.Name,
+		}, nil
+	case *storagepb.ExprDef_Duration:
+		return logicalplan.Duration(time.Duration(e.Duration.Milliseconds) * time.Millisecond), nil
+	case *storagepb.ExprDef_Convert:
+		expr, err := ExprFromProto(e.Convert.Expr)
+		if err != nil {
+			return nil, err
+		}
+
+		t, err := protoTypeToArrow(e.Convert.Type)
+		if err != nil {
+			return nil, err
+		}
+
+		return &logicalplan.ConvertExpr{
+			Expr: expr,
+			Type: t,
+		}, nil
+	case *storagepb.ExprDef_If:
+		cond, err := ExprFromProto(e.If.Condition)
+		if err != nil {
+			return nil, err
+		}
+
+		then, err := ExprFromProto(e.If.Then)
+		if err != nil {
+			return nil, err
+		}
+
+		els, err := ExprFromProto(e.If.Else)
+		if err != nil {
+			return nil, err
+		}
+
+		return &logicalplan.IfExpr{
+			Cond: cond,
+			Then: then,
+			Else: els,
+		}, nil
+	default:
+		return nil, fmt.Errorf("unsupported expression type: %T", e)
+	}
+}
+
+func protoTypeToArrow(t storagepb.Type) (arrow.DataType, error) {
+	switch t {
+	case storagepb.Type_TYPE_FLOAT64:
+		return arrow.PrimitiveTypes.Float64, nil
+	default:
+		return nil, fmt.Errorf("unsupported type: %v", t)
+	}
+}
+
+func protoOpToLogicalOp(op storagepb.Op) (logicalplan.Op, error) {
+	switch op {
+	case storagepb.Op_OP_EQ:
+		return logicalplan.OpEq, nil
+	case storagepb.Op_OP_NOT_EQ:
+		return logicalplan.OpNotEq, nil
+	case storagepb.Op_OP_LT:
+		return logicalplan.OpLt, nil
+	case storagepb.Op_OP_LT_EQ:
+		return logicalplan.OpLtEq, nil
+	case storagepb.Op_OP_GT:
+		return logicalplan.OpGt, nil
+	case storagepb.Op_OP_GT_EQ:
+		return logicalplan.OpGtEq, nil
+	case storagepb.Op_OP_REGEX_MATCH:
+		return logicalplan.OpRegexMatch, nil
+	case storagepb.Op_OP_REGEX_NOT_MATCH:
+		return logicalplan.OpRegexNotMatch, nil
+	case storagepb.Op_OP_AND:
+		return logicalplan.OpAnd, nil
+	case storagepb.Op_OP_OR:
+		return logicalplan.OpOr, nil
+	case storagepb.Op_OP_ADD:
+		return logicalplan.OpAdd, nil
+	case storagepb.Op_OP_SUB:
+		return logicalplan.OpSub, nil
+	case storagepb.Op_OP_MUL:
+		return logicalplan.OpMul, nil
+	case storagepb.Op_OP_DIV:
+		return logicalplan.OpDiv, nil
+	case storagepb.Op_OP_CONTAINS:
+		return logicalplan.OpContains, nil
+	case storagepb.Op_OP_NOT_CONTAINS:
+		return logicalplan.OpNotContains, nil
+	default:
+		return logicalplan.OpUnknown, fmt.Errorf("unsupported op: %v", op)
+	}
+}
+
+func protoAggFuncToLogicalAggFunc(f storagepb.AggregationFunction_Type) (logicalplan.AggFunc, error) {
+	switch f {
+	case storagepb.AggregationFunction_TYPE_SUM:
+		return logicalplan.AggFuncSum, nil
+	case storagepb.AggregationFunction_TYPE_MIN:
+		return logicalplan.AggFuncMin, nil
+	case storagepb.AggregationFunction_TYPE_MAX:
+		return logicalplan.AggFuncMax, nil
+	case storagepb.AggregationFunction_TYPE_COUNT:
+		return logicalplan.AggFuncCount, nil
```

Comment:
the internal version of this has new options here, can you sync this? the proto will need a small update as well